contracts/serverless-v2/Jobs.sol

```diff
@@ -0,0 +1,500 @@
+
+    function supportsInterface(
+        bytes4 interfaceId
+    )
+        public
+        view
+        virtual
+        override(ERC165Upgradeable, AccessControlUpgradeable)
+        returns (bool)
+    {
+        return super.supportsInterface(interfaceId);
+    }
+
+    function _authorizeUpgrade(
+        address /*account*/
+    ) internal view override onlyRole(DEFAULT_ADMIN_ROLE) {}
+
+    //-------------------------------- Overrides end --------------------------------//
+
+    //-------------------------------- Initializer start --------------------------------//
+
+    error JobsZeroAddressAdmin();
+
+    function initialize(
+        address _admin,
+        Executors _executors,
+        address _paymentPoolAddress
+    ) public initializer {
+        if(_admin == address(0))
+            revert JobsZeroAddressAdmin();
+
+        __Context_init_unchained();
+        __ERC165_init_unchained();
+        __AccessControl_init_unchained();
+        __UUPSUpgradeable_init_unchained();
+
+        _grantRole(DEFAULT_ADMIN_ROLE, _admin);
+
+        executors = _executors;
+        jobCount = 0;
+        paymentPool = _paymentPoolAddress;
+    }
+
+    //-------------------------------- Initializer end --------------------------------//
+
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    IERC20 public immutable TOKEN;
+
+    /// @notice Maximum age of a valid signature, in seconds.
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    uint256 public immutable SIGN_MAX_AGE;
+
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    uint256 public immutable EXECUTION_BUFFER_TIME;
+
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    uint256 public immutable NO_OF_NODES_TO_SELECT;
+
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    uint256 public immutable EXECUTOR_FEE_PER_MS;
+
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    uint256 public immutable STAKING_REWARD_PER_MS;
+
+    Executors public executors;
+
+    address public paymentPool;
+
+    //-------------------------------- Admin methods start --------------------------------//
+
+    function setExecutorsContract(Executors _executors) external onlyRole(DEFAULT_ADMIN_ROLE) {
+        executors = _executors;
+    }
+
+    function setPaymentPool(address _paymentPoolAddress) external onlyRole(DEFAULT_ADMIN_ROLE) {
+        paymentPool = _paymentPoolAddress;
+    }
+
+    //-------------------------------- Admin methods end ----------------------------------//
+
+    //-------------------------------- Job start --------------------------------//
+
+    struct Job {
+        uint256 jobId;
+        uint256 deadline;   // in milliseconds
+        uint256 execStartTime;
+        address jobOwner;
+        uint256 executionTime;   // it stores the execution time for first output submitted only (in milliseconds)
+        uint8 outputCount;
```

Comment:
if you're really optimizing storage, owner should be just before this so they take up 1 slot, o/w just do uint256